### PR TITLE
add code lint

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### CODE REVIEW

File: txnkv/txn.go [hunk 1]
```diff
@@ -265,14 +280,18 @@ func (txn *Transaction) LockKeys(keys ...key.Key) error {
 	return nil
 }
 
+// Valid returns if the transaction is valid.
+// A transaction become invalid after commit or rollback.
```
Comment (rleungx) [inline txnkv/txn.go]:
```suggestion
// A transaction becomes invalid after commit or rollback.
```

File: txnkv/txn.go [hunk 2]
```diff
@@ -118,6 +121,10 @@ func (txn *Transaction) String() string {
 	return fmt.Sprintf("txn-%d", txn.startTS)
 }
 
+// Iter creates an Iterator positioned on the first entry that k <= entry's key.
+// If such entry is not found, it returns an invalid Iterator with no error.
+// It yields only keys that < upperBound. If upperBound is nil, it means the upperBound is unbounded.
+// The Iterator must be Closed after use.
```
Comment (rleungx) [inline txnkv/txn.go]:
```suggestion
// The Iterator must be closed after use.
```

File: Makefile [hunk 3]
```diff
@@ -3,3 +3,7 @@ default:
 
 test:
 	GO111MODULE=on go test ./...
+
+check:
+	GO111MODULE=off go get golang.org/x/lint/golint
```
Comment (rleungx) [inline Makefile]: How about using `golangci-lint`?

Comment (disksing) [reply Makefile]: Maybe later? I want it to work first and then improve.

**Pull request merged**